src/android_internal/tracing_service_proxy.cc

```diff
@@ -58,6 +62,14 @@ bool ReportTrace(const char* reporter_package_name,
   // possible.
   android::base::unique_fd fd(owned_trace_fd);
 
+  // 'ReportTrace' may be called before system boot is completed (e.g., when
```

Comment:
Do we actually need this?

From b/461765543 looks like waitForService will wait for the service to be up, so you don' tneed to wait for boot (or any precondition) as the wait below will do the right thing anyways